src/main/java/com/onegini/examples/service/TokenValidationResultMappingService.java

```diff
@@ -26,7 +26,7 @@ public RequestMapperRequest modifyRequest(final RequestMapperRequest request) {
   private void mapUserId(final RequestMapperRequest request) {
     final String requestUri = request.getRequestUri();
     final TokenValidationResult tokenValidationResult = request.getTokenValidationResult();
-    final String userId = tokenValidationResult.getReferenceId();
+    final String userId = tokenValidationResult.getSub();
 
     request.setRequestUri(requestUri.concat("/" + userId));
   }
```

Comment:
please add a method to map the Token Type (amr) to a header. Similar to mapScopes. e.g. `TOKEN_TYPE` as header name. Please note this is an array so in theory multiple types can be returned. Use a space separated string a header value to map this array.